query-engine/query-engine-node-api/src/node_drivers/engine.rs

```diff
@@ -0,0 +1,140 @@
+use crate::engine::QueryEngine;
+
+#[allow(unused_imports)]
+use napi::bindgen_prelude::Promise as JsPromise;
+
+use napi::{Env, JsFunction, JsObject, JsString, JsUnknown};
+use napi_derive::napi;
+
+pub struct NodejsFunctionContext {
+    env: Env,
+    pub query_raw: JsFunction,
+    pub query_raw_typed: JsFunction,
+    pub execute_raw: JsFunction,
+    pub execute_raw_typed: JsFunction,
+    pub version: JsFunction,
+}
+
+impl NodejsFunctionContext {
+    pub fn query_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise
+
+        Ok(())
+    }
+
+    pub fn query_raw_typed(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("SELECT * FROM users").unwrap();
+        let _promise: JsUnknown = self.query_raw_typed.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn execute_raw(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("EXECUTE ...").unwrap();
+        let _promise: JsUnknown = self.execute_raw.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn execute_raw_typed(&self) -> napi::Result<()> {
+        let arg = self.env.create_string("EXECUTE ...").unwrap();
+        let _promise: JsUnknown = self.execute_raw_typed.call(None, &[arg])?;
+        // TODO: cast JsUnknown to JsPromise, idk how yet :/
+
+        Ok(())
+    }
+
+    pub fn version(&self) -> napi::Result<String> {
+        let version_raw_str: JsUnknown = self.version.call::<JsUnknown>(None, &[])?;
+        let version_js_str: JsString = version_raw_str.try_into()?;
+        let version_str = version_js_str.into_utf8()?.into_owned()?;
+
+        Ok(version_str)
+    }
+}
+
+fn read_nodejs_functions<'a>(env: Env, ctx: JsObject) -> napi::Result<NodejsFunctionContext> {
+    let query_raw = ctx.get_named_property("queryRaw")?;
```

Comment:
I believe this can be refactored into some sort of struct with the `#[napi]` annotations, but this'll do for the time being.